app/protocols.ts

```diff
@@ -0,0 +1,130 @@
+import { z } from 'zod'
+
+interface Protocol {
+  name: string
+  category: string
+  subcategory: string
+  contract: string
+  address: string
+}
+
+class ProtocolDatabase {
+  private protocols: Protocol[] = []
+  private isLoaded = false
+
+  async loadData() {
+    if (this.isLoaded) return
+
+    try {
+      const response = await fetch(
+        'https://raw.githubusercontent.com/monad-developers/protocols/refs/heads/main/protocols.csv'
+      )
+      const csvText = await response.text()
+      
+      const lines = csvText.split('\n')
+      
+      for (let i = 1; i < lines.length; i++) {
+        const values = lines[i].split(',')
+        if (values.length >= 5) {
+          this.protocols.push({
+            name: values[0]?.trim() || '',
+            category: values[1]?.trim() || '',
+            subcategory: values[2]?.trim() || '',
+            contract: values[3]?.trim() || '',
+            address: values[4]?.trim() || '',
+          })
+        }
+      }
+      this.isLoaded = true
```

Comment:
Can you use the KV store instead to cache the protocol data? It doesn't make sense to use an in memory db in next.js